pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchConverter.java

```diff
@@ -123,7 +135,9 @@ public static Optional<RawMessage> rebatchMessage(RawMessage msg,
             payload.readerIndex(readerIndex);
             brokerMeta = payload.readSlice(brokerEntryMetadataSize + Short.BYTES + Integer.BYTES);
         }
-        MessageMetadata metadata = Commands.parseMessageMetadata(payload);
+        if (metadata == null) {
+            metadata = Commands.parseMessageMetadata(payload);
```

Comment:
is there a need to skip the message metadata if it's not parsed?
Doesn't `parseMessageMetadata` move the read position of the ByteBuf forward?

Reply:
fixed